### ISSUE TITLE
Feature request:  Allow the width of the terminal to be configurable. (Useful for large/ultrawide monitors)

### ISSUE THREAD

**Issue by fc-asad** [amezin/gnome-shell-extension-ddterm#129]
Thank you for this great extension.

It would be great to have a feature that allows the users configure the width of the terminal either in px or percentage. It would be particularly useful for those using large/ultrawide monitors. So that the terminal takes up defined space at the centre of the screen and not the entire screen.

**Comment by alexboutros** [comment]
Would also love to see this implemented.

**Comment by k0d4** [comment]
+1

**Comment by slugcat-dev** [comment]
This is a necessary feature addition in my opinion. It's really annoying having to look to the left all the time when using a ultrawide monitor. Implementation isn't that complicated. An additional one preference slider for the width in percent would be enough, align the terminal window to the center of the screen and thats it.

Regarding horizontal window resizing, do it like the "System Details" dialog in GNOME Settings in the "About" section, where the dialog window is still center aligned to the main window after risizing.

**Comment by amezin** [comment]
I will accept a pull request that implements this feature - but only if it wont break anything else.

I'm not going to work on this feature myself, at least in the foreseeable future.

**Comment by amezin** [comment]
Since I (and nobody else too) can't get configurable window width to work reliably, could this solve the problem? I.e. reducing the width of the terminal - but not window

![Image](https://github.com/user-attachments/assets/258cb9a9-a4fb-4e3f-a0e7-be0e2ed11fcc)

**Comment by stevenilsen123** [comment]
Missing this feature is like the ONE REASON I choose Guake over ddterm. Would love to see this working

**Comment by amezin** [comment]
https://github.com/ddterm/gnome-shell-extension-ddterm/issues/129#issuecomment-2870264295 implemented in [v63.2.0](https://github.com/ddterm/gnome-shell-extension-ddterm/releases/tag/v63.2.0).

**Comment by chapati** [comment]
The issue seems to be still here 

<img width="2766" height="891" alt="Image" src="https://github.com/user-attachments/assets/5661ca2e-08a6-4a85-aa59-d7fbdd15d29d" />

**Comment by amezin** [comment]
Which issue exactly?

Terminal width is reduced on the screenshot.

I see that it's misaligned (not centered) though - is this the issue you're talking about?

**Comment by chapati** [comment]
@amezin May be I do not understand something then. Are these gray empty areas on both sides expected to be present? Is there any way to make background be of the size of the terminal window?

**Comment by amezin** [comment]
> Are these gray empty areas on both sides expected to be present?

Yes https://github.com/ddterm/gnome-shell-extension-ddterm/issues/129#issuecomment-2870264295

> Is there any way to make background be of the size of the terminal window?

Background is matching the size of the window. One of the window's dimensions still takes full screen size. It's possible to make the sides transparent (https://github.com/ddterm/gnome-shell-extension-ddterm/pull/2023, also should be possible with CSS), but clicks on them still won't go through anyway.

**Comment by chapati** [comment]
@amezin Understood. Thank you for explanation.
